### PR TITLE
Fixes so much shit

### PR DESCRIPTION
**Pull request by Carbonhell**
Ports from #12357 to #12406 : 
1)cleans up pneumatic cannon afterattack proc slightly and lets you place it in lockers etc correctly
2)moves atmos helmet from atmos closet to the secure locker
3)makes flooring get qdelled
4)doesn't let you toggle helmet light while crit/dead anymore
5)fixes cakehat lighting
6)stops aliens roleplaying Ghost Whisperer
7)generalizes equip_to_slot proc so it isn't copypasted on monkeys and humans
8)fixes lizard sprout coming out of EVA helmet and similar helmets
9)makes revenant click code use ghost code instead of copypasting said code
10)prevents revenants from emping smeses for balance purposes
11)adds in brain login.dm file which was removed accidentally on rebase
12)fixes EVA helmet sprite to not be trasparent on the face

### CODE REVIEW

**Comment by Spacedong** [comment]
pretty sure EVA helmet sprite was supposed to be  transparent as a feature but I don't like that anyway so +1

**Comment by Carbonhell** [comment]
Yeah, it was a feature, but it was conflicting with lizard mutparts such as the sprout,so.

**Comment by Spacedong** [comment]
oh rip then, i didnt care much for it anyway

**Comment by nalarmg** [comment]
Some of this seems balance related, needs head input before merge.

**Comment by Nokington** [comment]
wut

**Comment by Carbonhell** [comment]
There is nothing balance related other than being able to put pneumatic cannons in closets/racks without shooting them,which is a bugfix

File: code/modules/clothing/head/misc_special.dm [hunk 1]
```diff
@@ -69,16 +70,18 @@
...
+		force = 3
+		AddLuminosity(brightness_on)
+		damtype = BURN
+		icon_state = "cake1"
 		SSobj.processing |= src
 	else
-		src.force = null
-		src.damtype = BRUTE
-		src.icon_state = "cake0"
+		src.force = 0
```
Comment (ghost) [inline code/modules/clothing/head/misc_special.dm]: > removing src in one line but not in the next
> 
> inconsistencies

my EYES

**Comment by nalarmg** [comment]
"10)prevents revenants from emping smeses for balance purposes" seems contrary to your statement, Carbon. I'd like the heads to review this to make sure everything's alright with that in mind.

**Comment by Carbonhell** [comment]
No clue,being able to empty a SMES which has 2 milions of power in 3-4 seconds sounds like a bug to me,since apcs can't be drained either

**Comment by Nebyab** [comment]
First off carbon, if you call it balance, its pretty clearly balance, otherwise you'd not have said balance. In any case, I'm alright with this. I love revenant, but it really shouldn't be used to dick over the entire station like it can be currently.

**Comment by Nebyab** [comment]
other fixes are good as well

**Comment by Nokington** [comment]
this doesnt really need head input its just bug fixes. It makes no sense that they can fuck over smes but not apc so get rid of them fucking smes

**Pull request merged**